src/main/java/io/fabric8/maven/docker/BuildMojo.java

```diff
@@ -83,8 +84,18 @@ protected void buildAndTag(ServiceHub hub, ImageConfiguration imageConfig)
     protected Date getReferenceDate() {
         return new Date();
     }
+
     private String determinePullPolicy(BuildImageConfiguration buildConfig) {
-        return buildConfig != null && buildConfig.getImagePullPolicy() != null ? buildConfig.getImagePullPolicy() : imagePullPolicy;
+        if (buildConfig != null) {
+            if (buildConfig.getImagePullPolicy() != null) {
+                return buildConfig.getImagePullPolicy();
+            } else if (buildConfig.getAutoPull() != null) {
+                return buildConfig.getAutoPull().equalsIgnoreCase("TRUE") ?
+                        ImagePullPolicy.IfNotPresent.name() :
+                        ImagePullPolicy.Never.name();
+            }
```

Comment:
It was a long time ago, but what actually why do we need both `imagePullPolicy` and `autoPull` as properties ? It looks like they are really doing the same and autoPull could always be interfered from an `imagePullPolicy`.